jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -677,13 +677,16 @@ ecma_builtin_helper_def_prop (ecma_object_t *obj_p, /**< object */
   prop_desc.value = value;
 
   prop_desc.is_writable_defined = true;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
   prop_desc.is_writable = writable;
 
   prop_desc.is_enumerable_defined = true;
   prop_desc.is_enumerable = enumerable;
 
   prop_desc.is_configurable_defined = true;
   prop_desc.is_configurable = configurable;
+#pragma GCC diagnostic pop
```

Comment:
_Bool (which bool unfolds into) works fine.
The problem here is that on platforms where bool unfolds to _Bool8 (nuttx and derivatives configured with CONFIG_C99_BOOL8 undefined), we've got bool being uint_8.

This is what compiler complains about (some bits get discarded when casting 8-bit wide value to a narrower bitfield).